lib/ts/recipe/passwordless/recipeImplementation.ts

```diff
@@ -1,295 +1,104 @@
+            return await webJsImplementation.doesEmailExist.bind(this)(input);
         },
-        getLoginAttemptInfo: function ():
-            | undefined
-            | {
-                  deviceId: string;
-                  preAuthSessionId: string;
-                  flowType: FlowType;
-                  contactInfo: string;
-                  contactMethod: "EMAIL" | "PHONE";
-                  lastResend: number;
-                  redirectToPath?: string;
-              } {
-            const storedInfo = getLocalStorage(PASSWORDLESS_LOGIN_ATTEMPT_INFO_STORAGE_KEY);
-            if (!storedInfo) {
-                return undefined;
-            }
-            try {
-                const info = JSON.parse(storedInfo);
 
-                return {
-                    contactInfo: info.contactInfo,
-                    contactMethod: info.contactMethod,
-                    deviceId: info.deviceId,
-                    flowType: info.flowType,
-                    preAuthSessionId: info.preAuthSessionId,
-                    lastResend: info.lastResend,
-                    redirectToPath: info.redirectToPath,
-                };
-            } catch (ex) {
-                return undefined;
-            }
+        doesPhoneNumberExist: async function (input) {
+            return await webJsImplementation.doesPhoneNumberExist.bind(this)(input);
+        },
+        getLoginAttemptInfo: function <CustomAttemptInfoProperties>(input: { userContext: any }) {
+            return webJsImplementation.getLoginAttemptInfo.bind(this)<CustomAttemptInfoProperties>(input);
+        },
+        setLoginAttemptInfo: async function (input) {
+            return webJsImplementation.setLoginAttemptInfo.bind(this)<{
+                redirectToPath?: string;
+            }>({
+                attemptInfo: {
+                    ...input.attemptInfo,
+                    redirectToPath: getRedirectToPathFromURL(),
+                },
+                userContext: input.userContext,
+            });
         },
-        setLoginAttemptInfo: function (input: {
-            deviceId: string;
-            preAuthSessionId: string;
-            flowType: FlowType;
-            contactInfo: string;
-            contactMethod: "EMAIL" | "PHONE";
-            redirectToPath?: string;
-        }): void {
-            setLocalStorage(
-                PASSWORDLESS_LOGIN_ATTEMPT_INFO_STORAGE_KEY,
-                JSON.stringify({
-                    // This can make future changes/migrations a lot cleaner
-                    version: 1,
-                    ...input,
-                })
-            );
+        clearLoginAttemptInfo: function (input) {
+            return webJsImplementation.clearLoginAttemptInfo.bind(this)(input);
         },
-        clearLoginAttemptInfo: function (): void {
-            removeFromLocalStorage(PASSWORDLESS_LOGIN_ATTEMPT_INFO_STORAGE_KEY);
+        didLoginAttemptInfoChangeAfterResend: function (input) {
```

Comment:
- This seems to be unused
- I think you could drop "AfterResend" from the name.

Reply:
`didLoginAttemptInfoChangeAfterResend` will get removed (I just havent done it because I wanted to club review changes in one push)